app/latigo/time_series_api/client.py

```diff
@@ -171,13 +176,25 @@ def _get_metadata_from_api(self, name: str, facility: typing.Optional[str] = Non
                 ...
             }
         """
-        if not name:
+        if not tag_names:
             raise ValueError("No tag name is specified for fetching from Time Series API.")
 
-        body = {"name": name, "facility": facility} if facility else {"name": name}
-        res = self._get(self.base_url, params=body)
-        data = parse_request_json(res)
-        return self.get_only_item_from_metadata(metadata=data, tag_name=name)
+        gathered_tags_data = []
+        max_url_chars_amount = 8000
+
+        for batch in get_batches(tag_names, batch_size=max_tag_names_in_one_request):
+            names_to_fetch = "&".join((f"name={name}" for name in batch))
+            url = f"{self.base_url}?{names_to_fetch}{f'&facility={facility}' if facility else ''}"
+            if len(url) >= max_url_chars_amount:
+                raise ValueError(
+                    f"Length of the url to get Tags exceeded maximum allowed length and has '{len(url)}' chars."
+                )
+
+            res = self._get(url)
```

Comment:
There is no need to compound a URL stirng, `requests` accepts a list with `params` argument:
```suggestion
            params = {"name": batch}
            if facility:
                params["facility"] = facility
            res = self._get(url, params=params)
```

Reply:
You're right, `requests` does accept params, I thought that it'd be useful to have check that will throw an error with some details if somehow tags be named very largely, url will be too long and it'll make system crush. 
Now I think that we'll get the error from the `requests` anyway, so length of the url might be investigated with reproducing same model prediction from the logs of the error.
I'll adjust this.